examples/examples/subscribe_one_event.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is part of subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+//! To run this example, a local polkadot node should be running. Example verified against polkadot 0.9.13-82616422d0-aarch64-macos.
+//!
+//! E.g.
+//! ```bash
+//! curl "https://github.com/paritytech/polkadot/releases/download/v0.9.13/polkadot" --output /usr/local/bin/polkadot --location
+//! polkadot --dev --tmp
+//! ```
+
+use futures::{
+    future,
+    stream,
+    StreamExt,
+};
+use sp_keyring::AccountKeyring;
+use std::time::Duration;
+use subxt::{
+    ClientBuilder,
+    DefaultConfig,
+    DefaultExtra,
+    PairSigner,
+};
+
+#[subxt::subxt(runtime_metadata_path = "examples/polkadot_metadata.scale")]
+pub mod polkadot {}
+
+/// Subscribe to all events, and then manually look through them and
+/// pluck out the events that we care about.
+#[async_std::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    // Subscribe to any events that occur:
+    let api = ClientBuilder::new()
+        .build()
+        .await?
+        .to_runtime_api::<polkadot::RuntimeApi<DefaultConfig, DefaultExtra<DefaultConfig>>>();
+
+    // Subscribe to just balance transfer events, making use of `flat_map` and
+    // `filter_map` from the StreamExt trait to filter everything else out.
+    let mut transfer_events = api
+        .events()
+        .subscribe()
+        .await?
+        // Ignore errors returning events:
+        .filter_map(|events| future::ready(events.ok()))
+        // Map events to just those we care about:
+        .flat_map(|events| {
+            let transfer_events = events
+                .find::<polkadot::balances::events::Transfer>()
+                .collect::<Vec<_>>();
+            stream::iter(transfer_events)
+        });
```

Comment:
(I think one thing I started pondering was; how likely is it that people would care about plucking out more than one event type?)

Reply:
In most cases I would use such API since I only care about one event. However here is a case where I need two events: https://github.com/paritytech/cargo-contract/blob/0e9ffe78947f458c89c5ee0ef25234a146b797d9/src/cmd/extrinsics/instantiate.rs#L246.

Maybe tuples of event types would be one idea?

Reply:
Tuples would be the ideal way to provide the type, but the return type would need to be more sum-type in nature as we could get back 0, 1 or multiple of each event asked for. I have a couple of thoughts, but I think it'd be good to merge this first and add this functionality in a separate PR so that it can be given a little more thought!

Reply:
Absolutely! Defo not a job for this PR was just brainstorming here since it was brought up.

Reply:
Hehe, I suppose I was going to just add it if it was simple enough (ie the one-event variation)! Thanks for your thoughts; very useful :))